src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,32 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  setIsOpedModal: (newValue: boolean) => void;
+  selectedTodo: Todo | null;
+  setSelectedTodo: (newValue: Todo | null) => void;
+
+};
+
+export const TodoModal: React.FC<Props> = ({
+  setIsOpedModal,
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(selectedTodo?.userId || 0).then(setSelectedUser);
```

Comment:
Add `catch` block, also if you don't have `selectedTodo?.userId ` you should not do an API call